### PR TITLE
Add faraday middleware

### PR DESCRIPTION
**Pull request by yorikim**
Middleware add digest and signature

Sample usage:
```ruby
require 'faraday'
require 'faraday/http/signatures'
require 'pp'

conn = Faraday.new(url: 'http://localhost:9292') do |faraday|
  faraday.request :http_signatures, 'Test', File.read('spec/support/fixtures/rsa256/private.pem')
  faraday.adapter Faraday.default_adapter
end

response = conn.post('/') do |r|
  r.body = '{"hello": "world"}'
end
pp response
```

### CODE REVIEW

File: faraday-http-signatures.gemspec [hunk 1]
```diff
@@ -22,6 +22,6 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'rake', '~> 10.0'
   s.add_development_dependency 'rspec', '~> 3.0'
   s.add_development_dependency 'rubocop', '~> 0.39.0'
-  s.add_runtime_dependency 'rack', '>= 1.5'
-  s.add_runtime_dependency 'faraday', '>= 0.9.2'
+  s.add_runtime_dependency 'rack', '~> 1.5'
```
Comment (rjnienaber) [inline faraday-http-signatures.gemspec]: Why is ```rack``` a runtime dependency?

Comment (yorikim) [inline faraday-http-signatures.gemspec]: Forgot to remove it.

File: lib/faraday/http/signatures/digest_ciphers/sha256.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    module Signatures
+      module DigestCiphers
+        module SHA256
+          class << self
+            SHA256 = OpenSSL::Digest::SHA256.new
+
+            def encrypt(message)
+              SHA256.reset
+              SHA256 << message
+              SHA256.digest
```
Comment (rjnienaber) [inline lib/faraday/http/signatures/digest_ciphers/sha256.rb]: By making ```SHA256`` a class variable, doesn't this open it up to thread-safety issues?

File: lib/faraday/http/signatures/signature_ciphers/rsa256.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  module Http
+    module Signatures
+      module SignatureCiphers
+        class RSA256
+          class << self
+            DIGEST = OpenSSL::Digest::SHA256.new
+
+            def encrypt(key, data)
+              private = OpenSSL::PKey::RSA.new(key)
+              private.sign(DIGEST, data)
```
Comment (rjnienaber) [inline lib/faraday/http/signatures/signature_ciphers/rsa256.rb]: Even though this is allowed, I don't think we should be using variables that have the same name as Ruby keywords.

File: lib/faraday/http/signatures/digest_ciphers/sha256.rb (hunk 2 above)
Comment (yorikim) [inline lib/faraday/http/signatures/digest_ciphers/sha256.rb]: fixed

File: lib/faraday/http/signatures/signature_ciphers/rsa256.rb (hunk 3 above)
Comment (yorikim) [inline lib/faraday/http/signatures/signature_ciphers/rsa256.rb]: fixed https://github.com/CurrencyCloud/faraday-http-signatures/pull/4/commits/c07a7cf8becd818fb3c1c217555dbc8467a30658

**Pull request merged**